addons/account/models/sequence_mixin.py

```diff
@@ -303,7 +304,7 @@ def _set_next_sequence(self):
             format_values['seq'] = 0
             format_values['year'] = self._truncate_year_to_length(date_start.year, format_values['year_length'])
             format_values['year_end'] = self._truncate_year_to_length(date_end.year, format_values['year_end_length'])
-            format_values['month'] = date_start.month
+            format_values['month'] = self[self._sequence_date_field].month
```

Comment:
I tried to be as clear as possible in the commit message:
```
- Replace `date_start.month` by `self[self._sequence_date_field].month`,
  which seems more logical when reading the code -> We want to use the
  document month, not the fiscal year start month. It is actually the
  same value in all sequence cases (monthly, year, year range...) in the
  current code, but next commit will add a "year range monthly" case
  which will need this change.
```

I'll try to explain it better here, with my mind not being so fresh about it since it's bit quite some time.
This change basically does nothing at this point (before the next commits), it's the same "operation".

`self._sequence_date_field` = `date` (which is then using the `date` field of the `account.move` for instance).

First, when reading the code when starting this PR, this line bugged me out and I was wondering why were we using the sequence start month to name the account move rather than using the account move date month.

I dig into the code and figured it was always the same at the end, as the `month` format value is only used by the `_sequence_monthly_regex`, for which the sequence is only one month, and the account move date will always be in the same month as the sequence month obviously, so at the end using the month from the sequence or the account move date will be the same "by chance".

That explained why this line which looked wrong was actually not leading to any issue.

I kept it as is once I figured, but then I actually had to "fix" it because it wouldn't work "by chance" anymore with the new "year range monthly" format which is also now using the `month` format value but which should use the month of the account move, not the first month of the sequence, as the sequence is yearly (despite showing months too), it would always use the first month of the sequence (eg april) instead of the correct month of the account move.

I spent time figuring all this as I never opened any `account` code so every single line of code and concept were new to me, but hopefully it helps understanding more than what I wrote in the commit msg

Reply:
Thanks a lot for the extensive explanations, I got it now... :+1: